aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -188,6 +188,19 @@ Observe which of the `OperationId` values vary within a request, and between req
 
 * *Singleton* objects are the same for every object and every request (regardless of whether an instance is provided in `ConfigureServices`)
 
+## Scope validation
+
+When the app is running in the Development environment on ASP.NET Core 2.0 or later, the default service provider performs checks to verify that:
+
+* Scoped services aren't directly or indirectly resolved from the root service provider.
+* Scoped services aren't directly or indirectly injected into singletons.
+
+The root service provider is created when [BuildServiceProvider](/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectioncontainerbuilderextensions.buildserviceprovider) is called. The root service provider's lifetime usually corresponds to the app/server's lifetime.
+
+Scoped services are disposed by the container that created them. If a scoped service is created in the root container, the service's lifetime is effectively promoted to singleton because it's only disposed by the root container when app/server is shut down. Validating service scopes catches these situations when `BuildServiceProvider` is called.
```

Comment:
Consider removing the word "effectively". It doesn't seem to add any value.

Reply:
I think it adds some value because it wouldn't be correct to say that the "lifetime *IS* promoted to singleton." It it "*effectively* promoted to singleton" by virtue of the lifetime of the root container.

Reply:
I've read this a few times. We can't honestly say that it **IS** promoted to singleton when it's only happening by virtue of the timing of its disposal. If there's another wording to consider, I'm all :ear: :ear: :ear:. I like the sentence tho ... it's clear what the problem is the way that @pakrym stated it.